### ISSUE TITLE
Slash (/) in layer name causes tfjs converter error

### ISSUE THREAD

**Issue by aguang1201** [tensorflow/tfjs#89]
#### Describe the problem or feature request
1,I trained a keras model with densenet.
2,transfer the keras model with comand:tensorflowjs_converter --input_format keras \
3,saved the transfered the model dir on https://console.cloud.google.com/storage,and share the all of the model file.(include:model.json and lots of groupN files.
4,load the model into TensorFlow.js by providing the URL:
import * as tf from '@tensorflow/tfjs';
const model = await tf.loadModel('https://storage.googleapis.com/xxx/xxx_dir/model.json');

error is :
Failed to load https://storage.googleapis.com/xxx/xxx_dir/model.json: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:1234' is therefore not allowed access. The response had HTTP status code 400. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

How can I fix it with:response.headers.add('Access-Control-Allow-Origin', '*')?

Thank you .

**Comment by aguang1201** [comment]
For Enable CORS via the `Access-Control-Allow-Origin` header.
I used the command:node_modules/http-server/bin/http-server dist --cors
But it not valid

**Comment by nsthorat** [comment]
Ah, GCS, you should do this via gsutil:

You should create a "cors.json" somewhere with these contents:

```
[
  {
    "origin": ["*"],
    "responseHeader": ["Content-Type"],
    "method": ["GET", "HEAD"],
    "maxAgeSeconds": 3600
  }
]
```

Once you do that, run:

`gsutil cors set cors.json gs://your-bucket-name`

**Comment by aguang1201** [comment]
@nsthorat 
Thank you very much.
I solved this problem, but another problem came up．
When I transfered the keras model with comand:tensorflowjs_converter --input_format keras
The follow errors :
KeyError: "Can't open attribute (can't locate attribute: 'weight_names')"
It looks like a version issue.But I can not resolve it.
Could you help me?
Anyway thank you very much．
And Your speech is cool

**Comment by nsthorat** [comment]
Is this on the JavaScript side? Could you show us your model.json?

@caisq

**Comment by caisq** [comment]
@aguang1201 Showing us the model.json would be helpful. It would also be helpful for us to know what version of keras you are using.

**Comment by aguang1201** [comment]
@nsthorat @caisq 
It is not on JavaScript side.It is on keras side.
Before load the model into TensorFlow.js,I converted my pretrained Keras model to TF.js Layers format.
I used the Python API to export directly to TF.js Layers format:
**tfjs.converters.save_keras_model(model, tfjs_target_dir)**
##################################################################
The following errors came up:
Traceback (most recent call last):
  File "/home/panotech/PycharmProjects/tfjs-examples/chexnet_tfjs/python/convert_tfjs_model.py", line 20, in <module>
    convert()
  File "/home/panotech/PycharmProjects/tfjs-examples/chexnet_tfjs/python/convert_tfjs_model.py", line 17, in convert
    tfjs.converters.save_keras_model(model, 'model/tfjs_target_dir')
  File "/usr/local/share/anaconda3/envs/zzg3.6/lib/python3.6/site-packages/tensorflowjs/converters/keras_h5_conversion.py", line 263, in save_keras_model
    converter.h5_merged_saved_model_to_tfjs_format(temp_h5_path))
  File "/usr/local/share/anaconda3/envs/zzg3.6/lib/python3.6/site-packages/tensorflowjs/converters/keras_h5_conversion.py", line 162, in h5_merged_saved_model_to_tfjs_format
    layer, [name for name in layer.attrs['weight_names']])
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "/usr/local/share/anaconda3/envs/zzg3.6/lib/python3.6/site-packages/h5py/_hl/attrs.py", line 60, in __getitem__
    attr = h5a.open(self._id, self._e(name))
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py/h5a.pyx", line 77, in h5py.h5a.open
**KeyError: "Can't open attribute (can't locate attribute: 'weight_names')"**

Process finished with exit code 1
################################################################
keras version:2.1.5
h5py:2.8.0rc1
tensorflowjs:0.1.0

**Comment by aguang1201** [comment]
I made a mistake when I first tried.
I should convert the Keras model(graph+weights) to TF.js Layers format,But I used the weights filed only.
Until I saw the <"modelTopology": null> in model.json,I realize this error.
So this time I tried to convert the Keras model(graph+weights) to TF.js Layers format.
Then met the above error．

**Comment by caisq** [comment]
@aguang1201 

Can you try the following? Save your model as a .h5 file using for example:
```python
model.save('/tmp/model.h5')
```

Then use the binary that comes with the tensorflowjs pip package to do the conversion
```bash
tensorflowjs_converter --input_format keras /tmp/model_h5 /tmp/tfjs_model
```

If this works, let me know. If this doesn't work, can you shre the model.h5 file with me so I can take a look? Thanks.

**Comment by aguang1201** [comment]
@caisq 
Yes,I had saved the model with command:model.save('/tmp/model.h5'),
and pip install tensorflowjs and convert the model with command:
tensorflowjs_converter --input_format keras /tmp/model_h5 /tmp/tfjs_model
I tried it both in python2.7 and python3.6,but it doesn't work.
I will send model to you.Could you give me your email?
Thank you.

**Comment by caisq** [comment]
@aguang1201 My email address is the my github user name excluding the last character, followed by "at google dot com".

**Comment by aguang1201** [comment]
@caisq 
I shared the model file on google cloud.Please down it with follow link:
https://drive.google.com/file/d/10v1HAxB3KCvW-L9-idmfsj6Dyb8xgJa-/view?usp=sharing
Thank you

**Comment by caisq** [comment]
Thanks a lot for sharing the file with me, @aguang1201 

I quickly looked at your file and I noticed that one of your layer, called 'conv1', does not have any weights. It looks like a convolutional layer, so normally it ought to have some weights. Did you do something special with the layer?

**Comment by aguang1201** [comment]
@caisq 
I used the keras.applications.densenet.DenseNet121 to create the densenet model.
I have debugged into keras.applications.densenet.
DenseNet121.There is not a layer called 'conv1'.
But there is a layer called 'conv1/conv'.
The source is like:
x = Conv2D(64, 7, strides=2, use_bias=False, name='conv1/conv')(x)
x = BatchNormalization(axis=bn_axis, epsilon=1.001e-5,name='conv1/bn')(x)
x = Activation('relu', name='conv1/relu')(x)

**Comment by caisq** [comment]
@aguang1201 

Thank you again for the information. It is very helpful in helping us identify the root cause of the problem. I have come up with the following minimal reproduction code:

```python
import keras
import tensorflowjs as tfjs

model = keras.Sequential()
model.add(keras.layers.Dense(10, input_shape=[100], name='foo/bar'))

tfjs.converters.save_keras_model(model, '/tmp/tfjs_slash')
# --> Errors out
```

I'll see if I can get this fixed today and publish a patch release for `tensorflowjs` on PyPI.

**Comment by caisq** [comment]
@aguang1201 FYI. The bug should be fixed in the latest version (0.1.1) of tensorflowjs pip wheel:

https://pypi.python.org/pypi/tensorflowjs/0.1.1

**Comment by aguang1201** [comment]
@caisq 
Very efficient.
I verified the version (0.1.1), no errors.
Thank you very much
:+1:
